service-base/src/main/java/org/eclipse/hono/service/quarkus/OpenTelemetryProducer.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.service.quarkus;
+
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.quarkus.arc.Priority;
+
+/**
+ * A default producer for the OpenTelemetry instance.
+ * <p>
+ * This producer is an alternative of higher priority to the corresponding producer in <em>quarkus-opentelemetry</em>,
+ * and is not depending on OpenTelemetry usage to be enabled. This ensures there is always an OpenTelemetry bean
+ * available, potentially a no-op one if OpenTelemetry usage is disabled.
```

Comment:
can't we simply always enable OpenTelemetry?

Reply:
We could do that. This would simplify things here. My initial thought was to let the OpenTelemetry support be explicitly enabled via the `opentelemetry` maven profile (similar to the `jaeger` maven profile). 

But I guess the overhead of having OpenTelemetry always enabled is probably not that relevant anyway. 
Tracing could then be effectively disabled by using an empty [quarkus.opentelemetry.tracer.exporter.otlp.endpoint](https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry-exporter-otlp_quarkus.opentelemetry.tracer.exporter.otlp.endpoint) config property.

Reply:
> But I guess the overhead of having OpenTelemetry always enabled is probably not that relevant anyway.
Tracing could then be effectively disabled by using an empty [quarkus.opentelemetry.tracer.exporter.otlp.endpoint](https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry-exporter-otlp_quarkus.opentelemetry.tracer.exporter.otlp.endpoint) config property.

I think so too. And given that OpenTelemetry is the the designated standard to export tracing information ...

Reply:
Ok, OpenTelemetry is now always enabled and there is no `opentelemetry` maven profile any more.